airbyte-webapp/src/components/ui/Toast/Toast.tsx

```diff
@@ -1,86 +1,64 @@
 import React from "react";
-import styled, { keyframes } from "styled-components";
 
-import { H5 } from "components/base/Titles";
+import { CrossIcon } from "components/icons/CrossIcon";
+import { Text } from "components/ui/Text";
 
 import { Button } from "../Button";
-import { ErrorSign } from "./ErrorSign";
 import styles from "./Toast.module.scss";
 
-interface ToastProps {
-  title: string | React.ReactNode;
-  text?: string | React.ReactNode;
-  hasError?: boolean;
-  onClose?: () => void;
+export const enum ToastType {
+  WARNING = "warning",
+  SUCCESS = "success",
+  ERROR = "error",
+  INFO = "info",
 }
 
-export const SlideUpAnimation = keyframes`
-  0% {
-    translate(-50%, -100%);
-    bottom: -49px;
-  }
-  100% {
-    translate(-50%, 0);
-    bottom: 49px;
-  }
-`;
-
-const Singleton = styled.div<{ hasError?: boolean }>`
-  position: fixed;
-  bottom: 49px;
-  left: 50%;
-  transform: translate(-50%, 0);
-  z-index: 20;
-
-  padding: 25px 25px 22px;
-
-  background: ${({ theme, hasError }) => (hasError ? theme.lightDangerColor : theme.lightPrimaryColor)};
-  border: 1px solid ${({ theme }) => theme.greyColor20};
-  box-shadow: 0 1px 2px ${({ theme }) => theme.shadowColor};
-  border-radius: 8px;
-
-  display: flex;
-  flex-direction: row;
-  align-items: center;
-
-  animation: ${SlideUpAnimation} 0.25s linear;
-`;
-
-const Title = styled(H5)<{ hasError?: boolean }>`
-  color: ${({ theme, hasError }) => (hasError ? theme.dangerColor : theme.primaryColor)};
-
-  font-style: normal;
-  font-weight: bold;
-  font-size: 15px;
-  line-height: 18px;
-`;
+export interface ToastProps {
+  text: string | React.ReactNode;
+  type?: ToastType;
+  onAction?: () => void;
+  actionBtnText?: string;
+  onClose?: () => void;
+}
 
-const Text = styled.div`
-  color: ${({ theme }) => theme.mediumPrimaryColor};
+const ICON_MAPPING = {
+  [ToastType.WARNING]: faExclamation,
+  [ToastType.ERROR]: faTimes,
+  [ToastType.SUCCESS]: faCheck,
+  [ToastType.INFO]: faExclamation,
+};
 
-  font-style: normal;
-  font-weight: normal;
-  font-size: 14px;
-  line-height: 17px;
-  margin-top: 5px;
-`;
+const STYLES_BY_TYPE: Readonly<Record<ToastType, string>> = {
+  [ToastType.WARNING]: styles.warning,
+  [ToastType.ERROR]: styles.error,
+  [ToastType.SUCCESS]: styles.success,
+  [ToastType.INFO]: styles.info,
+};
 
-export const Toast: React.FC<ToastProps> = (props) => (
-  <Singleton hasError={props.hasError}>
-    {props.hasError && <ErrorSign />}
-    <div>
-      <Title hasError={props.hasError}>{props.title}</Title>
-      {props.text && <Text>{props.text}</Text>}
+export const Toast: React.FC<ToastProps> = ({ type = ToastType.INFO, onAction, actionBtnText, onClose, text }) => {
+  return (
+    <div className={classNames(styles.toastContainer, STYLES_BY_TYPE[type])}>
+      <div className={classNames(styles.iconContainer)}>
+        <FontAwesomeIcon icon={ICON_MAPPING[type]} className={styles.toastIcon} />
+      </div>
+      <div>
```

Comment:
With the latest changes, this div needs to `align-self: center` so that it looks good for single line toasts.

The current storybook looks like this:
<img width="509" alt="image" src="https://user-images.githubusercontent.com/168664/205983603-c41e8912-f45c-45d7-8276-c0fc35f6d83e.png">


But with the style, it will center the text vertically and also  look good when there is a lot of text:
<img width="452" alt="image" src="https://user-images.githubusercontent.com/168664/205984336-5ae5b2b3-5708-4254-a439-215f9b287c6b.png">